Compilation in dev mode fails on a warning about using an uninitialized variable

### Tested versions

Reproducible in 4.3.dev (https://github.com/godotengine/godot/commit/d3352813ea44447bfbf135efdec23acc4d1d3f89)


### System information

Windows 11 - Godot 4.3.dev

### Issue description

When upgrading Vulkan SDK to version 1.3.275.0 I came across the following issue:

When I try to compile the `master` branch with the `use_mingw=yes platform=windows dev_mode=yes` SCons options set, it fails at around 93% and prints the following info:

```
[ 93%] =====
In file included from servers\rendering\dummy\storage\material_storage.h:34,
                 from servers\rendering\dummy\storage\material_storage.cpp:31:
In copy constructor 'ShaderCompiler::DefaultIdentifierActions::DefaultIdentifierActions(const ShaderCompiler::DefaultIdentifierActions&)',
    inlined from 'RendererDummy::MaterialStorage::MaterialStorage()' at servers\rendering\dummy\storage\material_storage.cpp:40:27:
./servers/rendering/shader_compiler.h:89:16: error: 'actions' is used uninitialized [-Werror=uninitialized]
   89 |         struct DefaultIdentifierActions {
      |                ^~~~~~~~~~~~~~~~~~~~~~~~
servers\rendering\dummy\storage\material_storage.cpp: In constructor 'RendererDummy::MaterialStorage::MaterialStorage()':
servers\rendering\dummy\storage\material_storage.cpp:39:50: note: 'actions' declared here
   39 |         ShaderCompiler::DefaultIdentifierActions actions;
      |                                                  ^~~~~~~
cc1plus.exe: all warnings being treated as errors

=====
[ 98%] scons: *** [servers\rendering\dummy\storage\material_storage.windows.editor.x86_64.o] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:06:23.054]
```

Not sure if this information is relevant to this issue, but this is emitted when compilation starts:

```
py -3.9 -m SCons use_mingw=yes platform=windows dev_mode=yes
```

```
scons: Reading SConscript files ...
Auto-detected 8 CPU cores available for build parallelism. Using 7 cores by default. You can override it with the -j argument.
Using MinGW, arch x86_64
Building for platform "windows", architecture "x86_64", target "editor".
Checking for C header file mntent.h... (cached) no
```

### Steps to reproduce

- Try to build the source from the commit mentioned above with the following options set:
  - `use_mingw=yes platform=windows dev_mode=yes`


### Minimal reproduction project (MRP)

N/A

--- a/servers/rendering/shader_compiler.h
+++ b/servers/rendering/shader_compiler.h
@@ -52,38 +52,38 @@ public:
 		HashMap<StringName, bool *> usage_flag_pointers;
 		HashMap<StringName, bool *> write_flag_pointers;
 
-		HashMap<StringName, ShaderLanguage::ShaderNode::Uniform> *uniforms;
+		HashMap<StringName, ShaderLanguage::ShaderNode::Uniform> *uniforms = nullptr;
 	};
 
 	struct GeneratedCode {
 		Vector<String> defines;
 		struct Texture {
 			StringName name;
-			ShaderLanguage::DataType type;
-			ShaderLanguage::ShaderNode::Uniform::Hint hint;
+			ShaderLanguage::DataType type = ShaderLanguage::DataType::TYPE_VOID;
+			ShaderLanguage::ShaderNode::Uniform::Hint hint = ShaderLanguage::ShaderNode::Uniform::Hint::HINT_NONE;
 			bool use_color = false;
-			ShaderLanguage::TextureFilter filter;
-			ShaderLanguage::TextureRepeat repeat;
-			bool global;
-			int array_size;
+			ShaderLanguage::TextureFilter filter = ShaderLanguage::TextureFilter::FILTER_DEFAULT;
+			ShaderLanguage::TextureRepeat repeat = ShaderLanguage::TextureRepeat::REPEAT_DEFAULT;
+			bool global = false;
+			int array_size = 0;
 		};
 
 		Vector<Texture> texture_uniforms;
 
 		Vector<uint32_t> uniform_offsets;
-		uint32_t uniform_total_size;
+		uint32_t uniform_total_size = 0;
 		String uniforms;
 		String stage_globals[STAGE_MAX];
 
 		HashMap<String, String> code;
 
-		bool uses_global_textures;
-		bool uses_fragment_time;
-		bool uses_vertex_time;
-		bool uses_screen_texture_mipmaps;
-		bool uses_screen_texture;
-		bool uses_depth_texture;
-		bool uses_normal_roughness_texture;
+		bool uses_global_textures = false;
+		bool uses_fragment_time = false;
+		bool uses_vertex_time = false;
+		bool uses_screen_texture_mipmaps = false;
+		bool uses_screen_texture = false;
+		bool uses_depth_texture = false;
+		bool uses_normal_roughness_texture = false;
 	};
 
 	struct DefaultIdentifierActions {
@@ -91,8 +91,8 @@ public:
 		HashMap<StringName, String> render_mode_defines;
 		HashMap<StringName, String> usage_defines;
 		HashMap<StringName, String> custom_samplers;
-		ShaderLanguage::TextureFilter default_filter;
-		ShaderLanguage::TextureRepeat default_repeat;
+		ShaderLanguage::TextureFilter default_filter = ShaderLanguage::TextureFilter::FILTER_DEFAULT;
+		ShaderLanguage::TextureRepeat default_repeat = ShaderLanguage::TextureRepeat::REPEAT_DEFAULT;
 		int base_texture_binding_index = 0;
 		int texture_layout_set = 0;
 		String base_uniform_string;

